avocado/core/runners/sysinfo.py

```diff
@@ -0,0 +1,188 @@
+import multiprocessing
+import os
+import time
+import traceback
+
+from avocado.core import nrunner
+from avocado.core.runners.utils import messages
+from avocado.utils import software_manager
+from avocado.utils import sysinfo as sysinfo_collectible
+
+
+class PreSysInfo:
+    """
+    Log different system properties before start event.
+
+    An event may be a job, a test, or any other event with a
+    beginning and end.
+    """
+    sysinfo_dir = os.path.join('sysinfo', 'pre')
+
+    def __init__(self, config, sysinfo_config, queue):
+        """
+        Set sysinfo collectibles.
+
+        :param config: avocado configuration
+        :type config: dict
+        :param sysinfo_config: dictionary with commands/tasks which should be
+                              performed during the sysinfo collection.
+        :type sysinfo_config: dict
+        :param queue: queue for the runner messages
+        :type queue: multiprocessing.SimpleQueue
+        """
+        self.config = config
+        self.queue = queue
+        self._installed_pkgs = None
+        packages_namespace = 'sysinfo.collect.installed_packages'
+        self.log_packages = self.config.get(packages_namespace)
+        self.timeout = self.config.get('sysinfo.collect.commands_timeout')
+        self.locale = self.config.get('sysinfo.collect.locale')
+
+        self.sysinfo_config = sysinfo_config
+        self.collectibles = set()
+
+    def _set_collectibles(self):
+        for cmd in self.sysinfo_config.get("commands", []):
+            self.collectibles.add(
+                sysinfo_collectible.Command(cmd, timeout=self.timeout,
+                                            locale=self.locale))
+
+        for filename in self.sysinfo_config.get("files", []):
+            self.collectibles.add(sysinfo_collectible.Logfile(filename))
+
+    def _get_installed_packages(self):
+        sm = software_manager.SoftwareManager()
+        installed_pkgs = sm.list_all()
+        self._installed_pkgs = installed_pkgs
```

Comment:
And beyond @willianrampazzo point, I don't see a user for the protected `self._installed_pkgs` attribute.